design-proposals/network-binding-plugin/network-binding-plugin.md

```diff
@@ -257,6 +258,65 @@ VMI status and in one case to pass the data (processed) to the
 
 - DownwardAPI
 
+##### Device Info
+
+In some cases the multus status annotation
+contains [device-info](https://github.com/k8snetworkplumbingwg/device-info-spec/blob/main/SPEC.md).
+This info may be valuable for the binding plugin sidecar. For example
+`SRIOV` and `VDPA` plugins need this info.
+
+As the multus status annotation is not there yet on `virt-launcher` pod
+creation, the suggested solution is to use downwardAPI volume and mount
+it to the sidecar.
+The downwardAPI volume will track a new `virt-launcher` pod
+annotation `kubevirt.io/network-info` that will contain the `network-info`
+of the networks that have `downwardAPI`.
+Very similar to the
+existing [SRIOV downwardAPI](https://github.com/kubevirt/kubevirt/pull/8226)
+solution which is mapping the networkName to the `pci-address` of the
+device and mounting the downwardAPI volume to the `virt-launcher` pod.
+
+**Note**: Multus provides device-info information since version v3.7.
+If an older version is used, the downward API is not supported,
+the annotation will have empty values.
+
+Alternatives:
+
+- Instead of configuring in the kubevirt CR whether deviceInfo should be
+  mounted to the sidecar or not, we can pass via the downward API the device
+  info of all the interfaces that have it.
+  The downside in this approach is that a DownwardAPI volume will be added to
+  the pod even when it is not actually needed.
+- Passing `PCIDEVICE_<RESOURCE_NAME>_NET_INFO` env var from the
+  `virt-launcher` to the sidecar by adding it as an annotation to the
+  vmi object passed to the onDefineDomain method. The problem with this
+  approach is that it will reintroduce the
+  same [bug](https://github.com/kubevirt/kubevirt/issues/7444)
+  that the SRIOV downward API solved.
+- Sharing
+[CNIDeviceInfo](https://github.com/k8snetworkplumbingwg/device-info-spec/blob/main/SPEC.md#5-new-cni-plugin-arguments)
+file with the sidecar. It is a very similar approach to the
+downwardAPI option, the disadvantage here is that the user would have
+to specify the file path in the `NetworkAttachmentDefinition`.
+- Instead of using an annotation and DownwardAPI, extend the VMI interface status
+API to contain the interface device info.
+
+  Pros:
+  - DownwardAPI volume is not required, the VMI is already passed to the virt-launcher
+  compute and sidecar containers.
+
+  Cons:
+  - The VMI CRD should be changed.
+    The device infromation is exposed to users as a contract,
+    while the pod annotation with the DownwardAPI is an implementation detail with no such contract.
+  - It should be easy to move from the pod annotation to VMI Status in the future,
+  but we cannot go the opposite way.
+  - Keeping pod specific information in the VMI status may be raceful over migration.
+  - Users that have read permissions on the VMI will be able to see the device info.
```

Comment:
User that has read permission on the VMI not necessary has read permissions on the pod, therefore, with the downward API option, this kind of user won't be able to see the device info.

Reply:
Similarly, could it be possible that some users dont have access to VMI read but have access to read pods? 

All I am saying is that depending on the cluster configuration, some could have read access and other could not. We cannot make an assumption that one solution is more secure than other because it is largely dependent on cluster configuration

Reply:
We can say that the suggested option doesn't give any user more sensitive information than it already have.
The device info was already exposed to the pod via an environment variable by the device plugin. The downward API is needed just to create the mapping between the device info to the network using it.
But the virt-launcher is already familiar with the networks and with the device info.

Reply:
That sounds like an accurate statement, +1